python/ray/serve/_private/proxy_state.py

```diff
@@ -284,8 +284,15 @@ def is_drained(self) -> ProxyWrapperCallStatus:
         finished, _ = ray.wait([self._is_drained_obj_ref], timeout=0)
         if finished:
             self._is_drained_obj_ref = None
-            ray.get(finished[0])
-            return ProxyWrapperCallStatus.FINISHED_SUCCEED
+            is_drained = ray.get(finished[0])
+            if is_drained:
+                return ProxyWrapperCallStatus.FINISHED_SUCCEED
+            else:
+                # NOTE: Even though call returned successfully, we have to
```

Comment:
please update the docstring above to match this

Reply:
Not sure if there's much value since this is just a temporary cherry-pick and it's gonna be replaced by https://github.com/ray-project/ray/pull/41722/files anyway